test/e2e/auth/metadata_concealment.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	batch "k8s.io/api/batch/v1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = SIGDescribe("Metadata Concealment", func() {
+	f := framework.NewDefaultFramework("metadata-concealment")
+	completions := int32(1)
+
+	It("should run a check-metadata-concealment job to completion", func() {
+		framework.SkipUnlessProviderIs("gce")
+		By("Creating a job")
+		job := &batch.Job{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "check-metadata-concealment",
+			},
+			Spec: batch.JobSpec{
+				Template: v1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "check-metadata-concealment",
+					},
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{
+							{
+								Name:  "check-metadata-concealment",
+								Image: "gcr.io/google_containers/check-metadata-concealment:v0.0.1",
+							},
+						},
+						RestartPolicy: v1.RestartPolicyOnFailure,
```

Comment:
Don't we want this to restart never? Isn't failure how your signal that the test didn't work? Or do you just wait for the test to timeout?

Reply:
I want to restart in case we hit a race condition and schedule the job e.g. before the proxy is up.  So, yes, wait for the test to timeout.